### ISSUE TITLE
Vsphere provider, changing network name(label) recreates a VM instead of just updating

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-vsphere#331]
_This issue was originally opened by @marekmrowca as hashicorp/terraform#16953. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi,
When I am changing network name terraform will destroy existing VMs and create them again, instead of just update and restart.
Is there a workaround?
Is there a change it could be fixed in the next version?

```
terraform -v
Terraform v0.11.1
+ provider.null v1.0.0
+ provider.vsphere v0.4.2
```

```
-/+ vsphere_virtual_machine.chef-node[1] (new resource required)
      id:                                     "Terraform/mmtest/mmtest1" => <computed> (forces new resource)
      ...
      network_interface.#:                    "1" => "1"
      network_interface.0.adapter_type:       "vmxnet3" => "vmxnet3"
      network_interface.0.ip_address:         "" => <computed>
      network_interface.0.ipv4_address:       "*.*.*.*" => <computed>
      network_interface.0.ipv4_gateway:       "*.*.*.*" => <computed>
      network_interface.0.ipv4_prefix_length: "*" => <computed>
      network_interface.0.ipv6_address:       "" => <computed>
      network_interface.0.ipv6_gateway:       "" => <computed>
      network_interface.0.ipv6_prefix_length: "0" => <computed>
      network_interface.0.key:                "4000" => <computed>
      network_interface.0.label:              "MyNetwork" => "AnotherNetwork" (forces new resource)
      ...
```

Please help on this topic,
Best regards,
Marek

**Comment by vancluever** [comment]
Hey @marekmrowca, this was fixed in 1.0.0.

The interface has significantly changed since 0.4.2, so when you upgrade please review the [new resource documentation](https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html) for information on what needs to change.

Thanks!

**Comment by marekmrowca** [comment]
You're right. After the upgrade, changing networks works instantly.
Thank you!
